src/kdc/kdc_util.c

```diff
@@ -1444,6 +1407,41 @@ is_client_db_alias(krb5_context context, const krb5_db_entry *entry,
     return is_self;
 }
 
+/* Return true if realm canonicalizes to the same realm as entry->princ. */
+krb5_error_code
+is_realm_db_alias(krb5_context context, const krb5_db_entry *entry,
+                  krb5_data *realm, krb5_boolean *alias)
+{
+    krb5_error_code code;
+    krb5_principal alias_krbtgt;
+    krb5_db_entry *alias_entry;
+
+    if (data_eq(*realm, entry->princ->realm)) {
+        *alias = TRUE;
+        return 0;
+    }
+
+    code = krb5_build_principal_ext(context, &alias_krbtgt,
+                                    realm->length, realm->data,
+                                    KRB5_TGS_NAME_SIZE, KRB5_TGS_NAME,
+                                    realm->length, realm->data, 0);
+    if (code)
+        return code;
+
+    code = krb5_db_get_principal(context, alias_krbtgt, 0, &alias_entry);
+    krb5_free_principal(context, alias_krbtgt);
+    if (code) {
+        if (code != KRB5_KDB_NOENTRY)
+            return code;
+        *alias = FALSE;
+        return 0;
+    }
+
+    *alias = krb5_realm_compare(context, entry->princ, alias_entry->princ);
```

Comment:
Note: I think this would be better as "data_eq(alias_entry->princ->data[1], entry->princ->realm);"